tests/ecosystem/upgrade/test_logging_upgrade.py

```diff
@@ -0,0 +1,201 @@
+    )
+    csv_after_upgrade = csv.get(out_yaml_format=True)
+    clo_phase_after_upgrade = (
+        csv_after_upgrade.get('items')[0].get('status').get('phase')
+    )
+    eso_phase_after_upgrade = (
+        csv_after_upgrade.get('items')[1].get('status').get('phase')
+    )
+    if clo_phase_after_upgrade and eso_phase_after_upgrade == 'Succeeded':
+        logger.info("CSV phase Succeeded")
+        return True
+
+    logger.info("Upgrade not completed yet")
+    return False
+
+
+def check_version_after_upgrade():
+    """
+    Checks the version after upgrade
+
+    Returns:
+        bool: True if CSV version is correct after upgrade, False otherwise
+
+    """
+    csv = ocp.OCP(
+        kind=constants.CLUSTER_SERVICE_VERSION,
+        namespace=constants.OPENSHIFT_LOGGING_NAMESPACE
+    )
+    csv_after_upgrade = csv.get(out_yaml_format=True, wait=30)
+    clo_version_after_upgrade = (
+        csv_after_upgrade.get('items')[0].get('spec').get('version')
+    )
+    eso_version_after_upgrade = (
+        csv_after_upgrade.get('items')[1].get('spec').get('version')
+    )
+    if config.UPGRADE['upgrade_logging_channel'] in (
+        clo_version_after_upgrade and eso_version_after_upgrade
+    ):
+        logger.info(
+            f"Upgraded version of Cluster-logging "
+            f"operator {clo_version_after_upgrade}"
+        )
+        logger.info(
+            f"Upgraded version of Elastic-search "
+            f"operator {eso_version_after_upgrade}"
+        )
+        return True
+    return False
+
+
+@post_ocp_upgrade
+@pytest.mark.polarion_id("OCS-2201")
+class TestUpgradeLoggingAfterOCPUpgrade():
+    """
+    This class contains test for upgrade openshift-logging
+    after OCP upgrade
+    1. Monitors OCS health
+    2. Checks OCP version
+    3. Checks logging version, if mismatched then upgrades logging
+    4. Checks Elasticsearch and clusterlogging operator
+    5. Checks logging health post upgrade
+
+    """
+
+    def test_upgrade_logging(self):
+        """
+        This function contains test to upgrade openshift-logging
+        with Entry and Exit criteria for checks
+
+        """
+
+        ceph_cluster = CephCluster()
+        with CephHealthMonitor(ceph_cluster):
+
+            #  Pre-check
+            check_cluster_logging()
+
+            # Matching the OCP version and cluster-Logging version
+            ocp_version = get_ocp_version()
+            logger.info(f"OCP version {ocp_version}")
+            csv = ocp.OCP(
+                kind=constants.CLUSTER_SERVICE_VERSION,
+                namespace=constants.OPENSHIFT_LOGGING_NAMESPACE
+            )
+            csv_info = csv.get(
+                out_yaml_format=True
+            )
+            clo_version_before_upgrade = (
+                csv_info.get('items')[0].get('spec').get('version')
+            )
+            eso_version_before_upgrade = (
+                csv_info.get('items')[1].get('spec').get('version')
```

Comment:
After OCP upgrade we directly see both CSVs? Do you have the output of
```
oc get csv -n OPENSHIFT_LOGGING_NAMESPACE
```
Please replace the OPENSHIFT_LOGGING_NAMESPACE with relevant one.
Just curious how they are looking and don't have such cluster after upgrade or in upgrade process.

If I will see it for what we are looking for exactly I will maybe better understand it.

Reply:
Elasticsearch  is deployed on all namespaces by default so you will see 2 CSVs in openshift-logging as well.